pennylane/resource/resource.py

```diff
@@ -159,3 +388,17 @@ def _count_resources(tape) -> Resources:
             num_gates += 1
 
     return Resources(num_wires, num_gates, gate_types, gate_sizes, depth, shots)
+
+
+def _add_shots(s1: Shots, s2: Shots) -> Shots:
+    if s1.total_shots is None:
+        return s2
+
+    if s2.total_shots is None:
+        return s1
+
+    shot_vector = []
+    for shot in s1.shot_vector + s2.shot_vector:
+        shot_vector.append((shot.shots, shot.copies))
+
+    return Shots(shots=shot_vector)
```

Comment:
Looks good, I just wonder if it's ideal to have this defined here, or better in `pennylane/measurements/shots.py` as part of the `Shots` class.
Probably best to check with Soran.

Reply:
I was thinking the same thing, but I didn't want to add an unrelated file to the PR. I'll ask Soran.

Reply:
The current location looks good to me. @Jaybsoni please also check.